frontend/src/metabase/visualizations/components/settings/ChartSettingSeriesOrder.tsx

```diff
@@ -0,0 +1,160 @@
+  onChangeSeriesColor: (seriesKey: string, color: string) => void;
+}
+
+export const ChartSettingSeriesOrder = ({
+  onChange,
+  value: orderedItems,
+  onShowWidget,
+  hasEditSettings = true,
+  onChangeSeriesColor,
+}: ChartSettingSeriesOrderProps) => {
+  const [isSeriesPickerVisible, setSeriesPickerVisible] = useState(false);
+
+  const [visibleItems, hiddenItems] = useMemo(
+    () => _.partition(orderedItems, item => item.enabled),
+    [orderedItems],
+  );
+
+  const canAddSeries = hiddenItems.length > 0;
+
+  const toggleDisplay = useCallback(
+    (selectedItem: SortableItem) => {
+      const index = orderedItems.findIndex(
+        item => item.key === selectedItem.key,
+      );
+      onChange(updateIn(orderedItems, [index, "enabled"], enabled => !enabled));
+    },
+    [orderedItems, onChange],
+  );
+
+  const handleSortEnd = useCallback(
+    ({ id, newIndex }: DragEndEvent) => {
+      const oldIndex = orderedItems.findIndex(item => item.key === id);
+      onChange(arrayMove(orderedItems, oldIndex, newIndex));
+    },
+    [orderedItems, onChange],
+  );
+
+  const getItemTitle = useCallback((item: SortableItem) => {
+    return isEmpty(item.name) ? NULL_DISPLAY_VALUE : item.name;
+  }, []);
+
+  const handleOnEdit = useCallback(
+    (item: SortableItem, ref: HTMLElement | undefined) => {
+      onShowWidget(
+        {
+          props: {
+            seriesKey: item.key,
+          },
+        },
+        ref,
+      );
+    },
+    [onShowWidget],
+  );
+
+  const handleColorChange = useCallback(
+    (item: SortableItem, color: string) => {
+      onChangeSeriesColor(item.key, color);
+    },
+    [onChangeSeriesColor],
+  );
+
+  const handleAddSeries = useCallback(
+    (seriesKey: string) => {
+      const item = hiddenItems.find(item => item.key === seriesKey);
+      if (item) {
+        toggleDisplay(item);
+        setSeriesPickerVisible(false);
+      }
+    },
+    [hiddenItems, toggleDisplay],
+  );
+
+  const getId = useCallback((item: SortableItem) => item.key, []);
+
+  return (
+    <ChartSettingOrderedSimpleRoot>
+      {orderedItems.length > 0 ? (
+        <>
+          <ChartSettingOrderedItems
+            items={visibleItems}
+            getItemName={getItemTitle}
+            onRemove={visibleItems.length > 1 ? toggleDisplay : undefined}
+            onEnable={toggleDisplay}
+            onSortEnd={handleSortEnd}
+            onEdit={hasEditSettings ? handleOnEdit : undefined}
+            onColorChange={handleColorChange}
+            getId={getId}
+            removeIcon="close"
+          />
+          {canAddSeries && !isSeriesPickerVisible && (
+            <Button
+              variant="subtle"
+              onClick={() => setSeriesPickerVisible(true)}
+            >{t`Add another series`}</Button>
+          )}
+          {isSeriesPickerVisible && (
+            <Select
+              initiallyOpened
+              searchable
+              placeholder={t`Select a field`}
```

Comment:
Nit: maybe `Select a series`? Because when we are selecting breakout series they are not exactly fields
<img width="294" alt="Screenshot 2024-08-28 at 3 44 20 PM" src="https://github.com/user-attachments/assets/0e81fa81-0ecc-45c7-a9d0-e83995b99c88">